bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/TableProvider.java

```diff
@@ -204,7 +203,37 @@ private String generateCreateTableStatment(String tableName, BType constrainedTy
         return sb.toString();
     }
 
-    private String generateCreateTableStatment(String query, String newTableName) {
+    private void generateCreateTableStatement(int type, BArrayType sf, StringBuilder sb) {
+        switch (type) {
+            case TypeTags.INT_TAG:
+                sb.append(TableConstants.SQL_TYPE_BIGINT);
+                break;
+            case TypeTags.STRING_TAG:
+                sb.append(TableConstants.SQL_TYPE_VARCHAR);
+                break;
+            case TypeTags.FLOAT_TAG:
+            case TypeTags.DECIMAL_TAG:
+                sb.append(TableConstants.SQL_TYPE_DOUBLE);
+                break;
+            case TypeTags.BOOLEAN_TAG:
+                sb.append(TableConstants.SQL_TYPE_BOOLEAN);
+                break;
+            case TypeTags.JSON_TAG:
+            case TypeTags.XML_TAG:
+                sb.append(TableConstants.SQL_TYPE_CLOB);
+                break;
+            case TypeTags.ARRAY_TAG:
+                BType elementType = sf.getElementType();
+                if (elementType.getTag() == TypeTags.BYTE_TAG) {
+                    sb.append(TableConstants.SQL_TYPE_BLOB);
+                } else {
+                    sb.append(TableConstants.SQL_TYPE_ARRAY);
+                }
+                break;
```

Comment:
Shall we add a default case to throw an error here? In case there is a chance that nillable type with unsupported data type receives here